rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/handler/AnnotationHandlerChainBuilder.java

```diff
@@ -177,149 +129,8 @@ public ClassLoader run() {
-        } else if ("##XML_HTTP".equals(name)) {
-            name = "http://www.w3.org/2004/08/wsdl/http";
-        }
-        return name.contains(id);
-    }
-    private boolean patternMatches(Element el, QName comp) {
-        if (comp == null) {
-            return true;
-        }
-        String namePattern = el.getTextContent().trim();
-        if ("*".equals(namePattern)) {
-            return true;
-        }
-        final int idx = namePattern.indexOf(':');
-        if (idx < 0) {
-            String xml = StaxUtils.toString(el);
-            throw new WebServiceException(
-                BundleUtils.getFormattedString(BUNDLE,
-                                               "NOT_A_QNAME_PATTER",
-                                               namePattern, xml));
-        }
-        String pfx = namePattern.substring(0, idx);
-        String ns = el.lookupNamespaceURI(pfx);
-        if (ns == null) {
-            ns = pfx;
-        }
-        if (!ns.equals(comp.getNamespaceURI())) {
-            return false;
-        }
-        String localPart = namePattern.substring(idx + 1,
-                                                 namePattern.length());
-        if (localPart.contains("*")) {
-            //wildcard pattern matching
-            return Pattern.matches(mapPattern(localPart), comp.getLocalPart());
-        } else if (!localPart.equals(comp.getLocalPart())) {
-            return false;
-        }
-        return true;
-    }
-
-    private String mapPattern(String s) {
-        StringBuilder buf = new StringBuilder(s);
-        for (int x = 0; x < buf.length(); x++) {
-            switch (buf.charAt(x)) {
-            case '*':
-                buf.insert(x, '.');
-                x++;
-                break;
-            case '.':
-            case '\\':
-            case '^':
-            case '$':
-            case '{':
-            case '}':
-            case '(':
-            case ')':
-                buf.insert(x, '\\');
-                x++;
-                break;
-            default:
-                //nothing to do
-            }
-        }
-        return buf.toString();
-    }
-
-    private void processHandlerElement(Element el, List<Handler> chain) {
-        try {
-            JAXBContext ctx = getContextForPortComponentHandlerType();
-            PortComponentHandlerType pt = JAXBUtils.unmarshall(ctx, el, PortComponentHandlerType.class).getValue();
-            chain.addAll(buildHandlerChain(pt, classLoader));
-        } catch (JAXBException e) {
-            throw new IllegalArgumentException("Could not unmarshal handler chain", e);
-        }
-    }
-
-    private static synchronized JAXBContext getContextForPortComponentHandlerType()
-        throws JAXBException {
-        if (context == null) {
-            context = JAXBContext.newInstance(PortComponentHandlerType.class);
-        }
-        return context;
+    List<Handler> buildHandlerChain(PortComponentHandlerType ht) {
```

Comment:
I would prefer to keep this implementation details (either `private` or even remove the function altogether), but I am perfrectly fine to make the change after merge.

Reply:
It cannot be private as it is used by both JakartaeeHandlerChainBuilder & JavaeeHandlerChainBuilder.